### PR TITLE
Added Several Tags and Mechanisms

### PR DESCRIPTION
**Pull request by BehrRiley**
Adds 7 simple boolean tags and mechanisms. Rebased from my last broken PR

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCommandBlock.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import com.denizenscript.denizen.objects.MaterialTag;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.properties.Property;
+import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import org.bukkit.block.data.type.CommandBlock;
+
+public class MaterialCommandBlock implements Property {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCommandBlock.java]: class name doesn't match property

Comment (mcmonkey4eva) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCommandBlock.java]: Same problem on all other classes PR'd as well

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialGate.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    public static void registerTags() {
+
+        // <--[tag]
+        // @attribute <MaterialTag.signal_fire>
+        // @returns ElementTag(Boolean)
+        // @mechanism MaterialTag.signal_fire
+        // @group properties
+        // @description
+        // Returns whether this campfire will produce longer smoke trails, or not.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialGate.java]: This is just outright wrong copypastad meta

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCommandBlock.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    public static void registerTags() {
+
+        // <--[tag]
+        // @attribute <MaterialTag.conditional>
+        // @returns ElementTag(Boolean)
+        // @mechanism MaterialTag.conditional
+        // @group properties
+        // @description
+        // Returns whether this command block is conditional, or not.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCommandBlock.java]: Many of these seem like they'd easily fit into `MaterialMode`.

Comment (BehrRiley) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCommandBlock.java]: I'll merge this into `MaterialMode` in another set of changes I make later.

**Comment by mcmonkey4eva** [comment]
`Resolved Reviews`
Pressing the "Resolve" button doesn't actually resolve them (in the same way that minimizing your console doesn't fix errors).

**Comment by BehrRiley** [comment]
Closing for the moment - will PR again in the future when I have a bit more time to work on it.

**Pull request closed without merging**